File: facebook-hacker-cup/2011/studious-student/studious-student.js
const fs = require('fs')

let content = fs.readFileSync('studious-student.in', 'utf8')
content = content.split('\n').filter(c => c)

const N = content[0]
content.splice(0, 1)

for (let i = 0; i < N; i++) {
  const sentenceArray = content[i].split(' ')
  let M = sentenceArray[0]
  sentenceArray.splice(0, 1)
  for (let j = 0; j < M; j++) {
    for (let k = j + 1; k < M; k++) {
      let wordJ = sentenceArray[j] + sentenceArray[k]
      let wordK = sentenceArray[k] + sentenceArray[j]

      if (wordJ > wordK) {
        let temp = sentenceArray[j]
        sentenceArray[j] = sentenceArray[k]
        sentenceArray[k] = temp
      }
    }
  }

  console.log(`Case #${i + 1}: ${sentenceArray.join('')}`)
}

File: facebook-hacker-cup/2011/studious-student/studious-student.py
f = open('studious-student.in', 'r')
N = f.readline()
lines = f.readlines()

for index, line in enumerate(lines):
  line = line.split()
  M = int(line.pop(0))
  words = line
  for i in range(0, M):
    for j in range(i + 1, M):
      wordI = words[i] + words[j]
      wordJ = words[j] + words[i]
      if wordI > wordJ :
        words[i], words[j] = words[j], words[i]
  print("Case #{}: {}".format(index + 1, ''.join(words)))
